Too many open files

**Describe the bug**
```
2020/04/22 23:19:17 http: Accept error: accept tcp [::]:26657: accept4: too many open files; retrying in 320ms
I[2020-04-22|23:19:17.683] Ensure peers                                 module=pex numOutPeers=10 numInPeers=0 numDialing=0 numToDial=990
E[2020-04-22|23:19:18.692] Periodic WAL flush failed                    module=consensus wal=/home/pocket/.pocket/data/cs.wal/wal err="open /home/pocket/.pocket/data/cs.wal/wal: too many open files"
E[2020-04-22|23:19:18.692] Group's head may grow without bound          module=consensus wal=/home/pocket/.pocket/data/cs.wal/wal head=/home/pocket/.pocket/data/cs.wal/wal err="open /home/pocket/.pocket/data/cs.wal/wal: too many open files"
panic: open /home/pocket/.pocket/data/cs.wal: too many open files

goroutine 291 [running]:
github.com/tendermint/tendermint/libs/autofile.(*Group).readGroupInfo(0xc000494bb0, 0x0, 0x0, 0x0, 0x0)
        /home/pocket/go/pkg/mod/github.com/tendermint/tendermint@v0.32.9/libs/autofile/group.go:359 +0x547
github.com/tendermint/tendermint/libs/autofile.(*Group).checkTotalSizeLimit(0xc000494bb0)
        /home/pocket/go/pkg/mod/github.com/tendermint/tendermint@v0.32.9/libs/autofile/group.go:268 +0x64
github.com/tendermint/tendermint/libs/autofile.(*Group).processTicks(0xc000494bb0)
        /home/pocket/go/pkg/mod/github.com/tendermint/tendermint@v0.32.9/libs/autofile/group.go:239 +0x91
created by github.com/tendermint/tendermint/libs/autofile.(*Group).OnStart
        /home/pocket/go/pkg/mod/github.com/tendermint/tendermint@v0.32.9/libs/autofile/group.go:138 +0x66
```

**To Reproduce**
Run `pocket start`

**Expected behavior**
The server to keep running instead of panic.

**Additional context**
Reported by Danil Ushakov#5735 on Discord, running Pocket Core RC-0.2.3 on ubuntu 18.04 , 4vCPU, 8Gb RAM, 200Gb SSD. Attached is the `wal` file mentioned in the logs.

[wal.zip](https://github.com/pokt-network/pocket-core/files/4519082/wal.zip)

I have received a similar error:
```
E[2020-04-22|17:39:00.839] Failed to open WAL for consensus state       module=consensus wal=/Users/Germany/.pocket/data/cs.wal/wal err="open /Users/Germany/.pocket/data/cs.wal/wal: too many open files"
E[2020-04-22|17:39:00.839] Error loading ConsensusState wal             module=consensus err="open /Users/Germany/.pocket/data/cs.wal/wal: too many open files"
panic: Failed to start consensus state: open /Users/Germany/.pocket/data/cs.wal/wal: too many open files

conS:
ConsensusState

conR:
ConsensusReactor

goroutine 122 [running]:
github.com/tendermint/tendermint/consensus.(*ConsensusReactor).SwitchToConsensus(0xc00156b980, 0xa, 0x0, 0xc0015446f0, 0x6, 0xc0004492a0, 0x17, 0xe30, 0x1d6, 0xc00142adc0, ...)
	/Users/Germany/go/pkg/mod/github.com/tendermint/tendermint@v0.32.9/consensus/reactor.go:119 +0x271
github.com/tendermint/tendermint/blockchain/v0.(*BlockchainReactor).poolRoutine(0xc00041d6c0)
	/Users/Germany/go/pkg/mod/github.com/tendermint/tendermint@v0.32.9/blockchain/v0/reactor.go:271 +0x1192
created by github.com/tendermint/tendermint/blockchain/v0.(*BlockchainReactor).OnStart
	/Users/Germany/go/pkg/mod/github.com/tendermint/tendermint@v0.32.9/blockchain/v0/reactor.go:118 +0x84
``` 
i have tried to edit the syslog.conf file to add `ulimit -n 100000` on a mac but dealing with permission issues

--- a/app/config.go
+++ b/app/config.go
@@ -81,6 +81,8 @@ var (
 	pca *pocketCoreApp
 	// config
 	GlobalConfig Config
+	// HTTP CLIENT FOR TENDERMINT
+	tmClient *client.HTTP
 )
 
 type Config struct {
@@ -467,10 +469,14 @@ func privValState() {
 }
 
 func getTMClient() client.Client {
-	if GlobalConfig.PocketConfig.TendermintURI == "" {
-		return client.NewHTTP(DefaultTMURI, "/websocket")
+	if tmClient == nil {
+		if GlobalConfig.PocketConfig.TendermintURI == "" {
+			tmClient = client.NewHTTP(DefaultTMURI, "/websocket")
+		} else {
+			tmClient = client.NewHTTP(GlobalConfig.PocketConfig.TendermintURI, "/websocket")
+		}
 	}
-	return client.NewHTTP(GlobalConfig.PocketConfig.TendermintURI, "/websocket")
+	return tmClient
 }
 
 // get the hosted chains variable

